sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/caches/RxClientCollectionCache.java

```diff
@@ -80,27 +80,38 @@ public RxClientCollectionCache(ISessionContainer sessionContainer,
 
         request.getHeaders().put(HttpConstants.HttpHeaders.X_DATE, Utils.nowAsRFC1123());
 
-        String resourceName = request.getResourceAddress();
-        String authorizationToken = tokenProvider.getUserAuthorizationToken(
-                resourceName,
-                request.getResourceType(),
-                RequestVerb.GET,
-                request.getHeaders(),
-                AuthorizationTokenType.PrimaryMasterKey,
-                properties);
-
-        try {
-            authorizationToken = URLEncoder.encode(authorizationToken, "UTF-8");
-        } catch (UnsupportedEncodingException e) {
-            return Mono.error(new IllegalStateException("Failed to encode authtoken.", e));
+        if (tokenProvider.getAuthorizationTokenType() != AuthorizationTokenType.AadToken) {
+            String resourceName = request.getResourceAddress();
+            String authorizationToken = tokenProvider.getUserAuthorizationToken(
+                    resourceName,
+                    request.getResourceType(),
+                    RequestVerb.GET,
+                    request.getHeaders(),
+                    AuthorizationTokenType.PrimaryMasterKey,
+                    properties);
+
+            try {
+                authorizationToken = URLEncoder.encode(authorizationToken, "UTF-8");
+            } catch (UnsupportedEncodingException e) {
+                return Mono.error(new IllegalStateException("Failed to encode authtoken.", e));
+            }
+            request.getHeaders().put(HttpConstants.HttpHeaders.AUTHORIZATION, authorizationToken);
         }
-        request.getHeaders().put(HttpConstants.HttpHeaders.AUTHORIZATION, authorizationToken);
 
         if (retryPolicyInstance != null){
             retryPolicyInstance.onBeforeSendRequest(request);
         }
+
         Instant addressCallStartTime = Instant.now();
-        Mono<RxDocumentServiceResponse> responseObs = this.storeModel.processMessage(request);
+        Mono<RxDocumentServiceResponse> responseObs;
+        if (tokenProvider.getAuthorizationTokenType() != AuthorizationTokenType.AadToken) {
+            responseObs = this.storeModel.processMessage(request);
+        } else {
+            responseObs = tokenProvider
```

Comment:
Is special handling for performance?
Is there perf data to prove that it impacts performance?

For now NON-BLOCKER, but worth revisiting perf vs maintenance forward.